### PR TITLE
Feature request: allow extra options for storage backends

### PR DESCRIPTION
**Pull request by maxxiefjv**
I would like to allow some way of supporting a TLS connection, specifying the certificates. This is a draft PR meant for discussion

Edit:

As it is about a RedisWrapper this solution is probably better:

```python
    def __init__(self, db_uri, collection, ttl=None, options={}):
        if not _has_redis:
            raise ImportError("redis module is required but it is not available")
        self._db = Redis.from_url(db_uri, decode_responses=True)
        self._db = Redis.from_url(db_uri, decode_responses=True, **options)
        ...
```

### CODE REVIEW

File: src/pyop/storage.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ class RedisWrapper(StorageBase):
     Supports JSON-serializable data types.
     """
 
-    def __init__(self, db_uri, collection, ttl=None, options=None):
+    def __init__(self, db_uri, collection, ttl=None, options={}):
```
Comment (c00kiemon5ter) [inline src/pyop/storage.py:123]: that way we can recover if users set options to `None` explicitly

```suggestion
    def __init__(self, db_uri, collection, ttl=None, options={}):
        if options is None:
            options = {}
```

Comment (c00kiemon5ter) [reply src/pyop/storage.py:123]: hm, I think we can also do without this suggestion.

**Comment by c00kiemon5ter** [comment]
I think this looks ok

**Comment by maxxiefjv** [comment]
@c00kiemon5ter I updated the code to  your request. I agree that its best to handle this situation as most defaults appear to be None

File: src/pyop/storage.py [hunk 2]
```diff
@@ -120,10 +136,21 @@ class RedisWrapper(StorageBase):
     Supports JSON-serializable data types.
     """
 
-    def __init__(self, db_uri, collection, ttl=None):
+    def __init__(
+        self, db_uri, *, db_name=None, collection, ttl=None, extra_options=None
```
Comment (c00kiemon5ter) [inline src/pyop/storage.py:140]: Note: set all but the first arg to RedisWrapper to be keyword-only args.

Added `db_name` as the redis db index.

Comment (c00kiemon5ter) [reply src/pyop/storage.py:140]: Now, the MongoWrapper and RedisWrapper initialization interfaces are the same.

File: tests/pyop/test_storage.py [hunk 3]
```diff
@@ -69,15 +73,15 @@ def test_items(self, db):
...
         [
-            (["redis://localhost"], {"collection": "test"}),
-            (["redis://localhost", "test"], {}),
-            (["unix://localhost/0"], {"collection": "test", "ttl": 3}),
             (["mongodb://localhost/pyop"], {"collection": "test", "ttl": 3}),
             (["mongodb://localhost"], {"db_name": "pyop", "collection": "test"}),
             (["mongodb://localhost", "test", "pyop"], {}),
             (["mongodb://localhost/pyop", "test"], {}),
             (["mongodb://localhost/pyop"], {"db_name": "other", "collection": "test"}),
-            (["redis://localhost/0"], {"db_name": "pyop", "collection": "test"}),
```
Comment (c00kiemon5ter) [inline tests/pyop/test_storage.py:80]: `db_name` is now valid for redis.

**Pull request merged**